File: src/main/java/kitchenpos/products/tobe/ui/ProductRestController.java
package kitchenpos.products.tobe.ui;

import kitchenpos.products.tobe.application.ProductService;
import kitchenpos.products.tobe.domain.Product;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.net.URI;
import java.util.List;
import java.util.UUID;

@RequestMapping("/api/tobe/products")
@RestController("tobeProductRestController")
public class ProductRestController {

    private final ProductService productService;

    public ProductRestController(final ProductService productService) {
        this.productService = productService;
    }

    @PostMapping
    public ResponseEntity<Product> create(@RequestBody final ProductCreateRequest request) {
        final Product product = productService.create(request);
        return ResponseEntity.created(URI.create("/api/tobe/products/" + product.getId()))
                .body(product);
    }

    @PostMapping("/{productId}/price")
    public ResponseEntity<Product> changePrice(@PathVariable final UUID productId, @RequestBody final ProductChangePriceRequest price) {
        return ResponseEntity.ok(productService.changePrice(productId, price));
    }

    @GetMapping
    public ResponseEntity<List<Product>> findAll() {
        return ResponseEntity.ok(productService.findAll());
    }

    @PostMapping("/ids")
    public ResponseEntity<List<ProductResponse>> listByIds(@RequestBody final List<UUID> productIds) {
        return ResponseEntity.ok(productService.listByIds(productIds));
    }
}

File: src/main/java/kitchenpos/products/tobe/ui/ProductRestController.kt
package kitchenpos.products.tobe.ui

import kitchenpos.products.tobe.application.ProductService
import kitchenpos.products.tobe.ui.dto.CreateProductRequest
import kitchenpos.products.tobe.ui.dto.ProductResponse
import kitchenpos.products.tobe.ui.dto.UpdateProductPriceRequest
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import java.net.URI
import java.util.UUID

@RequestMapping("/api/products")
@RestController
class ProductRestController(
    private val productService: ProductService,
) {
    @PostMapping
    fun create(
        @RequestBody request: CreateProductRequest,
    ): ResponseEntity<ProductResponse> {
        val response = productService.create(request.name, request.price)
        return ResponseEntity.created(URI.create("/api/products/" + response.id))
            .body(response)
    }

    @PutMapping("/{productId}/price")
    fun changePrice(
        @PathVariable productId: UUID,
        @RequestBody request: UpdateProductPriceRequest,
    ): ResponseEntity<ProductResponse> {
        return ResponseEntity.ok(productService.changePrice(productId, request.price))
    }

    @GetMapping
    fun findAll(): ResponseEntity<List<ProductResponse>> {
        return ResponseEntity.ok(productService.findAll())
    }
}
